### PR TITLE
Add the ability to get Nid from MessageDigest and EcGroupRef

### PR DESCRIPTION
**Pull request by npmccallum**

### CODE REVIEW

File: openssl/src/hash.rs [hunk 1]
```diff
@@ -106,6 +106,11 @@ impl MessageDigest {
     pub fn size(&self) -> usize {
         unsafe { ffi::EVP_MD_size(self.0) as usize }
     }
+
+    /// The name of the digest
+    pub fn nid(&self) -> Nid {
```
Comment (sfackler) [inline openssl/src/hash.rs]: Can we call this `type_` to match with the openssl function name?

Comment (npmccallum) [reply openssl/src/hash.rs]: OpenSSL is generally inconsistent on their "give me the NID for this type" functions. I'd prefer that rust-openssl fixes this problem, especially given that their function names collide with Rust keywords.

**Comment by npmccallum** [comment]
Hrm... The build failure doesn't look related to my patches.

**Comment by npmccallum** [comment]
@sfackler Any update?

**Comment by sfackler** [comment]
Yeah, don't worry about that flake.

File: openssl/src/hash.rs (hunk 1 above)
Comment (sfackler) [reply openssl/src/hash.rs]: Our general policy is to follow the openssl names pretty directly.

**Comment by npmccallum** [comment]
@sfackler I updated the function names to match OpenSSL names as requested.

**Comment by sfackler** [comment]
Thanks!

**Comment by npmccallum** [comment]
@sfackler Do you have an ETA on a next release? I need these patches in my project, but building against git breaks my dependency tree.

**Comment by sfackler** [comment]
Done! You can use a Cargo patch to depend on a git dependency in a way that doesn't break the dependency graph for future reference: https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#working-with-an-unpublished-minor-version

**Pull request merged**